### PR TITLE
ui: fix adduser form

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

AddUser form wrongly shows the account dropdown when adding a user for an account. Account dropdown remains empty in the form.
When an account is not selected prior to opening the form, the account input should be required.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):
Without change:
Screenshot from 2021-06-02 00-14-03

With change:
Screenshot from 2021-06-02 00-14-47

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan ui

@rhtyd @utchoang @Pearl1594 Following code block is used in many forms (almost all) and when API is not accessible it will create an issue with `forEach` loop on params. Should we refactor this code-wide
```
  this.apiConfig = this.$store.getters.apis.API || {}
    this.apiConfig.params.forEach(param => {
      this.apiParams[param.name] = param
    })
```

**Comment by rhtyd** [comment]
@shwstppr we're very close to cutting RC1 - any change is acceptable if the scope is limited and it can be tested/merged with very high confidence.

**Comment by shwstppr** [comment]
@rhtyd let's consider the refactoring in master then. This change should fix the dysfunctional Add user form for now.

> @shwstppr we're very close to cutting RC1 - any change is acceptable if the scope is limited and it can be tested/merged with very high confidence.

**Comment by rhtyd** [comment]
+1
@shwstppr have you found any other forms which are failing? Maybe you can do a quick check, grep and see if you see other potential issues similar to this?

**Comment by Pearl1594** [comment]
@shwstppr I still see an issue while trying to add a user when we select the first account in the list 
image

As values.account points to the index and for the first account in the list index is 0, `if (values.account)` evaluates to false and the account parameter isn't set

**Comment by shwstppr** [comment]
@blueorangutan ui

File: ui/src/views/iam/AddUser.vue [hunk 1]
```diff
@@ -225,7 +227,7 @@ export default {
     this.apiConfig.params.forEach(param => {
       this.apiParams[param.name] = param
     })
-    this.apiConfig = this.$store.getters.apis.authorizeSamlSso || {}
+    this.apiConfig = this.$store.getters.apis.authorizeSamlSso || { params: [] }
```
Comment (shwstppr) [inline ui/src/views/iam/AddUser.vue:230]: **This change can be removed if #5064 is accepted**

**Comment by shwstppr** [comment]
@blueorangutan ui

**Comment by rhtyd** [comment]
@davidjumani @Pearl1594 can you review test this.

**Review by utchoang — APPROVED** [review]

**Pull request merged**